scarlet/blend.py

```diff
@@ -172,6 +173,9 @@ def fit(self, steps=200, e_rel=1e-2):
                     Ls=self._Ls, update=update, update_order=update_order, steps_g_update=steps_g_update, max_iter=steps,
                     e_rel=self._e_rel, e_abs=self._e_abs)
 
+            X, converged, errors = res
+            self.converged = all(converged)
```

Comment:
It would probably be useful to also store the convergence of each component, so `self.convergence=converged`. This might make it easier to troubleshoot in the future.

Reply:
In principle yes, but the ordering of that structure is not obvious outside of `Blend.fit`: it depends on the `update_order` keyword if its all components of `A` and then of `S` or the other way round.

I expressly don't want to make it a property of `Component` even though what you suggest is a per-component flag.

Reply:
that'll work. I'll unpack the convergence flags in `fit` and add a `has_converged(k)` method which spits out the a tuple for `(A,S)`.

Reply:
even better: I made `self.converged` a list of those tuples, ordered by `k`. It's very close to your original request and avoid the new method